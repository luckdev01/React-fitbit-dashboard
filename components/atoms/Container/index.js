import PropTypes from 'prop-types'

import spacing from '../../../styles/spacing'

export default function Container ({ children, max }) {
  return (
    <div className="container">
      {children}
      <style jsx>{`
        .container {
          width: 100%;
          max-width: ${max};
          padding: 0 ${spacing.small};
          margin: auto;
        }
      `}</style>
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.any,
  max: PropTypes.string,
}

Container.defaultProps = {
  max: '1024px',
}
