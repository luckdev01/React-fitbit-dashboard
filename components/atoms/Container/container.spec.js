import Container from './'

describe('Container', () => {
  it('renders children', () => {
    const child = 'this is a test'
    const wrapper = shallow(<Container>{child}</Container>)
    expect(wrapper).toMatchSnapshot()
  })
})
