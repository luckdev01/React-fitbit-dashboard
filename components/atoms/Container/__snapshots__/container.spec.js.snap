// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    max="1024px"
  >
    this is a test
  </Container>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "this is a test",
        <JSXStyle
          css=".container.__jsx-style-dynamic-selector{width:100%;max-width:1024px;padding:0 8px;margin:auto;}"
          dynamic={
            Array [
              "1024px",
              "8px",
            ]
          }
          styleId="391923204"
        />,
      ],
      "className": "jsx-4183663630 container",
    },
    "ref": null,
    "rendered": Array [
      "this is a test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".container.__jsx-style-dynamic-selector{width:100%;max-width:1024px;padding:0 8px;margin:auto;}",
          "dynamic": Array [
            "1024px",
            "8px",
          ],
          "styleId": "391923204",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "this is a test",
          <JSXStyle
            css=".container.__jsx-style-dynamic-selector{width:100%;max-width:1024px;padding:0 8px;margin:auto;}"
            dynamic={
              Array [
                "1024px",
                "8px",
              ]
            }
            styleId="391923204"
          />,
        ],
        "className": "jsx-4183663630 container",
      },
      "ref": null,
      "rendered": Array [
        "this is a test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".container.__jsx-style-dynamic-selector{width:100%;max-width:1024px;padding:0 8px;margin:auto;}",
            "dynamic": Array [
              "1024px",
              "8px",
            ],
            "styleId": "391923204",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
